agent/unit_state.go

```diff
@@ -71,12 +73,19 @@ type UnitStatePublisher struct {
 // them to the Registry every 5s. Heartbeat objects are also published as they
 // are received on the channel.
 func (p *UnitStatePublisher) Run(beatchan <-chan *unit.UnitStateHeartbeat, stop chan bool) {
+	var publishPeriod time.Duration
+	if p.ttl > 30*time.Second {
```

Comment:
Well, to me `sleepTime =10mins - 10secs` feeels very very very very (4x ;) wrong as well.

* We do a write to etcd with TTL 10 mins
* we sleep for 9mins 50secs
* We start the next action, might take more than 10secs
* ttl expires, we are lost

Why the deliberate high wait? I propbably just misunderstand some values here, but that sounds wrong.